src/main/kotlin/com/hibob/kotlinBasic/StoreService.kt

```diff
@@ -0,0 +1,47 @@
+package com.hibob.kotlinBasic
+
+import org.springframework.stereotype.Component
+import java.time.LocalDate
+import java.time.LocalDateTime
+
+@Component
+class StoreService {
+
+    fun pay(cart: List<Cart>, payment: Payment): Map<String, Check> {
+        return cart.map { cart -> cart.clientId to checkout(cart, payment) }.toMap()
+    }
+
+    fun checkout(cart : Cart , payment: Payment): Check {
+        val totalAmount = cart.products.filter { product -> check(product.custom) }.sumOf { product->product.price }
+
+        if (checkPayment(payment , totalAmount)) return Check(cart.clientId , Statuses.SUCCESS , totalAmount)
+        return Check(cart.clientId , Statuses.FAILURE , 0.0)
+    }
+
+
+
+    fun checkPayment (payment: Payment ,price : Double): Boolean {
+    return when(payment){
+        is Payment.Cash-> fail("you cant pay with cash")
+        is Payment.PayPal-> "@" in payment.email
+        is Payment.CreditCard-> (payment.type == CreditCardType.MASTERCARD || payment.type == CreditCardType.VISA ) && (payment.expiryDate.isAfter(LocalDate.now())) &&(price<payment.limit)&&(payment.number.length==10)
```

Comment:
this condition is not clear first I suggest you in case of CreditCard to call function called validateCreditCard.
then I would suggest you to split the condition to parts, and each part to use function that going to be more readable and easier to fix.

something like:


```
    fun checkPayment (payment: Payment ,price : Double): Boolean {
        return when(payment){
            is Payment.Cash-> fail("you cant pay with cash")
            is Payment.PayPal-> "@" in payment.email
            is Payment.CreditCard-> validateCreditCard(payment, price)
        }
    }

    private fun validateCreditCard(payment: Payment.CreditCard, price: Double) =
        isSupportedType(payment.type) && isCardDetailsValid(payment) && isNotOverTheLimit(price, payment)

    private fun isSupportedType(cardType: CreditCardType) =
        (cardType == CreditCardType.MASTERCARD || cardType == CreditCardType.VISA)

    private fun isCardDetailsValid(payment: Payment.CreditCard) =
        payment.expiryDate.isAfter(LocalDate.now())  && payment.number.length == 10

    private fun isNotOverTheLimit(price: Double, payment: Payment.CreditCard) = price < payment.limit
```

look at the condition now, I think its easier to read it